src/payment/coupon/data/sagas.js

```diff
@@ -18,18 +17,12 @@ export function* handleAddCoupon(action) {
   yield put(addCouponBegin());
   try {
     const result = yield call(postCoupon, action.payload.code);
-    yield put(fetchBasketSuccess(result));
+    yield call(handleBasketResult, result);
     yield call(handleMessages, result.messages, true);
```

Comment:
Should `handleMessages` just be part of `handleBasketResult`?

Reply:
We handle messages on success and errors on failure, but both need to populate the basket.